src/Symfony/Component/Security/Core/Encryption/SymmetricEncryption.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encryption;
+
+use Symfony\Component\Security\Core\Exception\DecryptionException;
+use Symfony\Component\Security\Core\Exception\MalformedCipherException;
+use Symfony\Component\Security\Core\Exception\UnsupportedAlgorithmException;
+
+/**
+ * Symmetric encryption uses the same key to encrypt and decrypt a message. The
+ * keys should be kept safe and should not be exposed to the public. The key length
+ * should be 32 bytes, but other sizes are accepted.
+ *
+ * Symmetric encryption is in theory weaker than asymmetric encryption.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class SymmetricEncryption
+{
+    /**
+     * @var string application secret
+     */
+    private $secret;
```

Comment:
I forget all the time, but can't we use property types for new features? 🧐

Reply:
Isn't that a 7.4 feature? We still need to support 7.2.

Reply:
I'm looking forward to Symfony 6. 😈